bootstrap/src/main/kotlin/jp/co/soramitsu/bootstrap/controller/EthController.kt

```diff
@@ -69,38 +70,91 @@ class EthController {
+    fun deployRelayRegistry(@NotNull @RequestBody request: EthereumNetworkProperties): ResponseEntity<DeploySmartContractResponse> {
+        return try {
+            val deployHelper = createSmartContractDeployHelper(request)
+            val relayRegistry = deployHelper.deployUpgradableRelayRegistrySmartContract()
+            deployHelper.web3.shutdown()
+            ResponseEntity.ok(DeploySmartContractResponse(relayRegistry.contractAddress))
+        } catch (e: Exception) {
+            log.error("Cannot deploy RelayRegistry smart contract", e)
+            ResponseEntity.status(HttpStatus.CONFLICT)
+                .body(DeploySmartContractResponse(e.javaClass.simpleName, e.message))
+        }
+    }
+
+    @PostMapping("/deploy/D3/masterContract")
+    fun deployMasterSmartContract(@NotNull @RequestBody request: DeployMasterContractRequest): ResponseEntity<DeployMasterContractResponse> {
+        return try {
+            val deployHelper = createSmartContractDeployHelper(request.network)
+            val master =
+                deployHelper.deployUpgradableMasterSmartContract(
+                    request.relayRegistryAddress,
+                    request.notaryEthereumAccounts
+                )
+            deployHelper.web3.shutdown()
+            ResponseEntity.ok(DeployMasterContractResponse(master.contractAddress, master.tokens.send()[0].toString()))
+        } catch (e: Exception) {
+            log.error("Cannot deploy RelayRegistry smart contract", e)
+            val response = DeployMasterContractResponse()
+            response.errorCode = e.javaClass.simpleName
+            response.message = e.message
+            ResponseEntity.status(HttpStatus.CONFLICT)
+                .body(response)
+        }
+    }
 
+    @PostMapping("/deploy/D3/relayImplementation")
+    fun deployRelayImplementation(@NotNull @RequestBody request: DeployRelayImplementationRequest): ResponseEntity<DeploySmartContractResponse> {
+        return try {
+            val deployHelper = createSmartContractDeployHelper(request.network)
+            val relayImplementation = deployHelper.deployRelaySmartContract(request.masterContractAddress)
+            deployHelper.web3.shutdown()
+            ResponseEntity.ok(DeploySmartContractResponse(relayImplementation.contractAddress))
+        } catch (e: Exception) {
+            log.error("Cannot deploy RelayImplementation smart contract", e)
+            ResponseEntity.status(HttpStatus.CONFLICT)
+                .body(DeploySmartContractResponse(e.javaClass.simpleName, e.message))
+        }
+    }
+
+    @PostMapping("/deploy/D3/smartContracts")
+    fun deployInitialSmartContracts(@NotNull @RequestBody request: AllInitialContractsRequest): ResponseEntity<DeployInitialContractsResponse> {
         log.info { "Run predeploy with notary addresses: ${request.notaryEthereumAccounts}" }
-        var response: MasterContractResponse
+        var response: DeployInitialContractsResponse
```

Comment:
do we need this variable? I mean we can declare `val` before the `return`

Reply:
yep